superset/reports/commands/execute.py

```diff
@@ -333,11 +334,25 @@ def _get_notification_content(self) -> NotificationContent:
                     error_text = "Unexpected missing screenshot"
             elif (
                 self._report_schedule.chart
-                and self._report_schedule.report_format == ReportDataFormat.DATA
+                and self._report_schedule.report_format == ReportDataFormat.CSV
+            ):
+                data = self._get_csv_data()
+                if not data:
+                    error_text = "Unexpected missing csv file"
+            elif (
+                self._report_schedule.chart
+                and self._report_schedule.report_format == ReportDataFormat.XLSX
             ):
                 csv_data = self._get_csv_data()
                 if not csv_data:
-                    error_text = "Unexpected missing csv file"
+                    error_text = "Unexpected missing csv data for xlsx"
+                else:
+                    df = pd.read_csv(BytesIO(csv_data))
+                    bio = BytesIO()
+                    # pylint: disable=abstract-class-instantiated
+                    with pd.ExcelWriter(bio, engine="openpyxl") as writer:
+                        df.to_excel(writer, index=False)
+                    data = bio.getvalue()
```

Comment:
I would recommend moving most of the csv and excel creation logic here to a new command called something like `_get_excel_data` so that it can be reusable, too.

Reply:
Done.